cucumber/api/features/rotate_api_key.feature

```diff
@@ -7,38 +7,153 @@ Feature: Rotate the API key of a role
+    When I PUT "/authn/cucumber/api_key?role=user:bob@test_policy" with username "carl" and password ":cucumber:user:carl_api_key"
+    Then the HTTP response status code is 401
+
+  Scenario: A User with update privilege CAN NOT rotate another user's API key using their password
+    Given I set the password for "carl" to "L0g-in-C@rl-123"
+    When I PUT "/authn/cucumber/api_key?role=user:bob@test_policy" with username "carl" and password "L0g-in-C@rl-123"
+    Then the HTTP response status code is 401
+
+# Host rotating own API key
+
+  Scenario: A Host CAN NOT rotate their own API key using an access token
+    Given I login as "host/host1"
+    When I PUT "/authn/cucumber/api_key"
+    Then the HTTP response status code is 401
+
+  Scenario: A Host CAN NOT rotate their own API key using an access token and a role parameter with self role value
+    Given I login as "host/host1"
+    When I PUT "/authn/cucumber/api_key?role=host:host1"
     Then the HTTP response status code is 401
 
-  Scenario: The API key can be rotated by foreign role when it has 'update' privilege
-    Given I create a new user "bob"
-    And I permit user "alice" to "update" user "bob"
-    And I login as "alice"
-    When I PUT "/authn/cucumber/api_key?role=user:bob"
-    Then the result is the API key for user "bob"
+  Scenario: A Host CAN rotate their own API key using their API key
+    When I PUT "/authn/cucumber/api_key?role=host:host1" with username "host/host1" and password ":cucumber:host:host1_api_key"
+    Then the HTTP response status code is 200
     And the HTTP response content type is "text/plain"
+    And the result is the API key for host "host1"
+
+# Host with update permission rotating another role's API key
+
+  Scenario: A Host with update privilege CAN rotate a user's API key with an access token
+    Given I login as "host/host1"
+    When I PUT "/authn/cucumber/api_key?role=user:carl"
+    Then the HTTP response status code is 200
+    And the HTTP response content type is "text/plain"
+    And the result is the API key for user "carl"
+
+  Scenario: A Host with update privilege CAN NOT rotate a user's API key with their own API key
+    When I PUT "/authn/cucumber/api_key?role=user:carl" with username "host/host1" and password ":cucucmber:host:host1_api_key"
+    Then the HTTP response status code is 401
+
+# Host without update permission rotating another role's API key
+
+  Scenario: A Host with update privilege CAN NOT rotate a user's API key with an access token
+    Given I login as "host/host2"
+    When I PUT "/authn/cucumber/api_key?role=user:carl"
+    Then the HTTP response status code is 401
+
+  Scenario: A Host without update privilege CAN NOT rotate a user's API key with their own API key
+    When I PUT "/authn/cucumber/api_key?role=user:carl" with username "host/host2" and password ":cucumber:host:host2_api_key"
+    Then the HTTP response status code is 401
```

Comment:
I like that you have covered all the possible cases but effectively the test I think is trying to cover way too much and due to the myriad of usernames that are unrelated to the test names will make this hard to maintain. 
To make it a bit more maintainable:
- I think we can remove the "owner" tests I think they are a superset of  "update" ones
- Maybe use usernames and hostnames indicative of their action (e.g. `delegated_user1`, `delegate_for_user1`)
- Add comments to the policy so it's obvious what each policy rule is doing
- Wrap common cucumber [groups](https://cucumber.io/docs/gherkin/reference/) of some sort
- Anything else that may make maintaining these tests easier for a newcomer